### PR TITLE
Create Signaling script (#1)

### PR DESCRIPTION
**Pull request by morph027**
initial support for signaling server, shoud fix https://github.com/nextcloud/vm/pull/1317

### CODE REVIEW

File: apps/talk_signaling.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    esac
+else
+    print_text_in_color "$ICyan" "Installing ${DESCRIPTION}..."
+fi
+
+# Install
+. /etc/lsb-release
+for package in nats-server nextcloud-spreed-signaling janus
+do
+  curl -sL -o "/etc/apt/trusted.gpg.d/morph027-${key}.asc" "https://packaging.gitlab.io/${key}/gpg.key"
```
Comment (enoch85) [inline apps/talk_signaling.sh:50]: We have a function called `curl_to_dir` which I think would fit here?

https://github.com/nextcloud/vm/blob/master/lib.sh#L306

Comment (morph027) [reply apps/talk_signaling.sh:50]: Sure, will add.

File: apps/talk_signaling.sh [hunk 2]
```diff
@@ -43,8 +45,6 @@ fi
 
 # Install
 . /etc/lsb-release
-is_process_running dpkg
```
Comment (morph027) [inline apps/talk_signaling.sh]: why did you remove this? not neccessary anymore? I've copied this over from another script.

Comment (enoch85) [reply apps/talk_signaling.sh]: I moved it to the top simce the first installation happens there.

File: apps/talk_signaling.sh (hunk 1 above)
Comment (enoch85) [reply apps/talk_signaling.sh:50]: :+1:

**Comment by enoch85** [comment]
Hey @morph027 Do you think we can book a session to fix this PR when we both are online?

**Comment by enoch85** [comment]
@szaimen You're also welcome to work on this. :smile:

**Comment by morph027** [comment]
Huh, sorry, was quite busy. I'll have some spare time for coding in 2 weeks time. Until then, i'm available for anything else (review, hints, ...).

**Comment by enoch85** [comment]
@morph027 OK, let's book a "hackathon" in 2 weeks then. I'll send you an invite.

We can use Talk (hehe) for chatting.

**Comment by enoch85** [comment]
hey @morph027, are you still up for the "hakathon" we booked?

**Comment by morph027** [comment]
> hey @morph027, are you still up for the "hakathon" we booked?

Yes ;) I'll have some spare time this evening too, if you like.

**Comment by enoch85** [comment]
@szaimen We will work on this today at 16:00 Swedish local time.

Please join if you want. We have created a Talk room here: https://cloud.hanssonit.se/call/yqqruj3m

**Comment by enoch85** [comment]
OK, I'm waiting for you in the Talk room.

cc @szaimen @morph027

**Comment by szaimen** [comment]
Hm yah, I am currently cooking. But I might join later.

**Comment by enoch85** [comment]
I'll grab something to eat. BRB.

File: apps/talk_signaling.sh [hunk 3]
```diff
@@ -33,7 +33,7 @@ lowest_compatible_nc 19
 
 ####################### TALK (COTURN)
 
-# Check if adminer is already installed
+# Check if talk/spreed is already installed
```
Comment (enoch85) [inline apps/talk_signaling.sh:36]: haha yea, saw that one too. :D

File: apps/talk_signaling.sh [hunk 4]
```diff
@@ -48,8 +48,8 @@ then
             occ_command_no_check config:app:delete spreed turn_servers
             occ_command_no_check config:app:delete spreed signaling_servers
             occ_command_no_check app:remove spreed
-            rm $TURN_CONF
-            apt-get purge coturn -y
+            rm "$TURN_CONF" "$SIGNALING_SERVER_CONF"
```
Comment (enoch85) [inline apps/talk_signaling.sh:51]: please add -f here.

File: apps/talk_signaling.sh [hunk 5]
```diff
@@ -48,8 +48,8 @@ then
...
+            apt-get purge coturn nats-server janus nextcloud-spreed-signaling -y
```
Comment (enoch85) [inline apps/talk_signaling.sh:52]: please add autoremove here.

File: apps/talk_signaling.sh [hunk 6]
```diff
@@ -59,8 +59,8 @@ then
             occ_command_no_check config:app:delete spreed turn_servers
             occ_command_no_check config:app:delete spreed signaling_servers
             occ_command_no_check app:remove spreed
-            rm $TURN_CONF
-            apt-get purge coturn -y
+            rm "$TURN_CONF" "$SIGNALING_SERVER_CONF"
```
Comment (enoch85) [inline apps/talk_signaling.sh:62]: `-f` please

File: apps/talk_signaling.sh [hunk 7]
```diff
@@ -59,8 +59,8 @@ then
...
+            apt-get purge coturn nats-server janus nextcloud-spreed-signaling -y
```
Comment (enoch85) [inline apps/talk_signaling.sh:63]: autoremove please

File: apps/talk_signaling.sh [hunk 8]
```diff
@@ -210,48 +210,74 @@ domain_check_200 "$SUBDOMAIN"
...
+sed -i "s|#turn_rest_api_key\s*=.*|$JANUS_API_KEY|" /etc/janus/janus.jcfg
+sed -i "s|#full_trickle|full_trickle|g" /etc/janus/janus.jcfg
+sed -i 's|#interface.*|interface = "lo"|g' /etc/janus/janus.transport.websockets.jcfg
+sed -i 's|#ws_interface.*|ws_interface = "lo"|g' /etc/janus/janus.transport.websockets.jcfg
 check_command systemctl restart janus
 check_command systemctl enable janus
 
+# HPB
+## Installation
+curl -sL -o "/etc/apt/trusted.gpg.d/morph027-nextcloud-spreed-signaling.asc" "https://packaging.gitlab.io/nextcloud-spreed-signaling/gpg.key"
```
Comment (enoch85) [inline apps/talk_signaling.sh:244]: Isn't this already fetched in the `for` loop?

Comment (enoch85) [reply apps/talk_signaling.sh:244]: Aah, saw now that you removed the loop...

**Comment by enoch85** [comment]
@morph027 please check comments

**Comment by enoch85** [comment]
we need to add 8081 to NONO_PORTS

File: apps/talk_signaling.sh [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+then
+    print_text_in_color "$IGreen" "$TURN_CONF was successfully created."
+else
+    print_text_in_color "$IRed" "Unable to create $TURN_CONF, exiting..."
+    print_text_in_color "$IRed" "Please report this issue here $ISSUES"
+    exit 1
+fi
+
+# Restart the TURN server
+check_command systemctl restart coturn.service
```
Comment (szaimen) [inline apps/talk_signaling.sh:174]: this has failed for me

File: apps/talk_signaling.sh [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+curl -sL -o "/etc/apt/trusted.gpg.d/morph027-nats-server.asc" "https://packaging.gitlab.io/nats-server/gpg.key"
+echo "deb [arch=amd64] https://packaging.gitlab.io/nats-server nats main" > /etc/apt/sources.list.d/morph027-nats-server.list
+apt update -q4 & spinner_loading
+install_if_not nats-server
+## Configuration
+mkdir -p /etc/nats
+install -d -o nats -g nats /etc/nats
+echo "listen: 127.0.0.1:4222" > /etc/nats/nats.conf
+chown nats:nats /etc/nats/nats.conf
+check_command systemctl restart nats-server
```
Comment (szaimen) [inline apps/talk_signaling.sh:238]: this has failed for me

File: apps/talk_signaling.sh [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+    a2ensite "$SUBDOMAIN.conf"
+    restart_webserver
+else
+    # remove settings to be able to start over again
+    rm -f "$HTTPS_CONF"
+    last_fail_tls "$SCRIPTS"/apps/talk_signaling.sh
+    
+    msg_box "Please run this script again to uninstall if you want clean system, or reinstall if you want to try again."
+    exit 1
+fi
```
Comment (szaimen) [inline apps/talk_signaling.sh:403]: hm, there is no combobox at the end?

Comment (enoch85) [reply apps/talk_signaling.sh:403]: combobox? sorry, I don't understand?

File: apps/talk_signaling.sh [hunk 12]
```diff
@@ -0,0 +1,403 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+NC_UPDATE=1 && TURN_INSTALL=1 . <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
```
Comment (szaimen) [inline apps/talk_signaling.sh:8]: BTW: I don't like it that you have to enter the domain at the very beginning of the script... maybe we can do it like in onlyoffice-docker?

File: apps/talk_signaling.sh [hunk 13]
```diff
@@ -0,0 +1,403 @@
...
+              /etc/janus \
+              /etc/apt/trusted.gpg.d/morph027-janus.asc \
+              /etc/apt/trusted.gpg.d/morph027-nats-server.asc \
+              /etc/apt/trusted.gpg.d/morph027-nextcloud-spreed-signaling.asc \
+              /etc/apt/sources.list.d/morph027-nextcloud-spreed-signaling.list\
+              /etc/apt/sources.list.d/morph027-janus.list \
+              /etc/apt/sources.list.d/morph027-nats-server.list
+            apt-get purge coturn nats-server janus nextcloud-spreed-signaling -y
+            apt autoremove -y
+            check_command systemctl disable signaling
```
Comment (szaimen) [inline apps/talk_signaling.sh:67]: this will fail if it is already uninstalled

File: apps/talk_signaling.sh (hunk 12 above)
Comment (enoch85) [reply apps/talk_signaling.sh:8]: Sure, please change.

File: apps/talk_signaling.sh (hunk 13 above)
Comment (enoch85) [reply apps/talk_signaling.sh:65]: Please remove

Comment (szaimen) [reply apps/talk_signaling.sh:65]: ive changed it

**Comment by enoch85** [comment]
```
root@cloud:/var/scripts# bash test.sh 
Site test.lindnercloud.com disabled.
To activate the new configuration, you need to run:
  systemctl reload apache2
Enabling site test.lindnercloud.com.
To activate the new configuration, you need to run:
  systemctl reload apache2
{"nextcloud-spreed-signaling":"Welcome","version":"0dcb9f6b2d9629ca4e41e78cd7aa61ff5d80f714"}
```

curl -L https://test.lindnercloud.com/api/v1/welcome

File: apps/talk_signaling.sh (hunk 12 above)
Comment (enoch85) [reply apps/talk_signaling.sh:8]: @szaimen  Could you please add this change to the new PR?
